File: mobile/concord/lib/api/models/token_response.dart
import 'json_utils.dart';

class TokenResponse {
  const TokenResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.accessTokenExpires,
    required this.refreshTokenExpires,
  });

  factory TokenResponse.fromJson(Map<String, dynamic> json) {
    return TokenResponse(
      accessToken: json.field('AccessToken') as String,
      refreshToken: json.field('RefreshToken') as String,
      accessTokenExpires: parseDateTime(json.field('AccessTokenExpires')),
      refreshTokenExpires: parseDateTime(json.field('RefreshTokenExpires')),
    );
  }

  final String accessToken;
  final String refreshToken;
  final DateTime accessTokenExpires;
  final DateTime refreshTokenExpires;

  Map<String, dynamic> toJson() => {
        'AccessToken': accessToken,
        'RefreshToken': refreshToken,
        'AccessTokenExpires': accessTokenExpires.toIso8601String(),
        'RefreshTokenExpires': refreshTokenExpires.toIso8601String(),
      };
}

File: backend/Concord/Concord.Application/Models/TokenResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class TokenResponse
{
    [JsonPropertyName("AccessToken")]
    public string? AccessToken { get; set; }

    [JsonPropertyName("RefreshToken")]
    public string? RefreshToken { get; set; }

    [JsonPropertyName("AccessTokenExpires")]
    public DateTime? AccessTokenExpires { get; set; }

    [JsonPropertyName("RefreshTokenExpires")]
    public DateTime? RefreshTokenExpires { get; set; }

    [JsonPropertyName("TwoFactorRequired")]
    public bool TwoFactorRequired { get; set; }

    [JsonPropertyName("TwoFactorToken")]
    public string? TwoFactorToken { get; set; }

    [JsonPropertyName("EmailConfirmationRequired")]
    public bool EmailConfirmationRequired { get; set; }
}